### PR TITLE
fix(queue): `FAILED_CONTINUE` tasks should not halt stage execution

### PR DESCRIPTION
**Pull request by ajordens**
The stage should continue executing even if a task completes with
the `FAILED_CONTINUE` status.

### CODE REVIEW

File: orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandlerTest.kt [hunk 1]
```diff
@@ -50,137 +50,83 @@ object CompleteTaskHandlerTest : SubjectSpek<CompleteTaskHandler>({
...
-      action("the handler receives a message") {
-        subject.handle(message)
-      }
-
-      it("updates the task state in the stage") {
-        verify(repository).storeStage(check {
-          it.tasks.first().apply {
-            assertThat(status).isEqualTo(SUCCEEDED)
-            assertThat(endTime).isEqualTo(clock.millis())
+  setOf(SUCCEEDED, FAILED_CONTINUE).forEach { successfulStatus ->
```
Comment (ajordens) [inline orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandlerTest.kt]: TBH this is the only change -- everything else is fallout from indenting this block

**Comment by ajordens** [comment]
Noticed when doing some local testing of the `AddServerGroupEntityTagsTask` that subsequent tasks were not being executed after `FAILED_CONTINUE` was returned.

I'm not sure we return that status very often (`AddServerGroupEntityTagsTask` and `CompleteCanaryTask` could do it).

**Comment by robfletcher** [comment]
I think historically we never returned that status directly, it was only applied after the fact when dealing with the end of the stage.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandler.kt [hunk 2]
```diff
@@ -49,7 +49,7 @@ class CompleteTaskHandler(
       } else {
         repository.storeStage(mergedContextStage)
 
-        if (message.status != SUCCEEDED) {
+        if (!setOf(SUCCEEDED, FAILED_CONTINUE).contains(message.status)) {
```
Comment (robfletcher) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandler.kt]: `message.status !in setOf(SUCCEEDED, FAILED_CONTINUE)`

File: orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandlerTest.kt (hunk 1 above)
Comment (robfletcher) [reply orca-queue/src/test/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandlerTest.kt]: Wish GitHub could ignore whitespace changes like IntelliJ can.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandler.kt (hunk 2 above)
Comment (ajordens) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteTaskHandler.kt]: done

**Pull request merged**